test/shared_test/lib_battery_powerflow_test.cpp

```diff
@@ -1593,3 +1995,307 @@ TEST_F(BatteryPowerFlowTest_lib_battery_powerflow, TestDCConnected)
+    EXPECT_NEAR(m_batteryPower->powerConversionLoss, 2.32, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemLoss, 0.5, error);
+    EXPECT_NEAR(m_batteryPower->powerCritLoadUnmet, 27.81, error);
+    EXPECT_NEAR(m_batteryPower->powerInterconnectionLoss, 0.0, error);
+
+    check_net_flows("4th case");
+
+
+    // discharge battery
+    m_batteryPower->powerBatteryDC = 20;
+    m_batteryPower->powerSystemLoss = 1.0;
+    m_batteryPowerFlow->calculate();
+
+    EXPECT_NEAR(m_batteryPower->powerBatteryAC, 17.47, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToLoad, 23.50, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToBattery, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToBattery, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToLoad, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToGrid, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerBatteryToLoad, 17.47, error);
+    EXPECT_NEAR(m_batteryPower->powerConversionLoss, 3.05, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemLoss, 1.0, error);
+    EXPECT_NEAR(m_batteryPower->powerCritLoadUnmet, 9.05, error);
+    EXPECT_NEAR(m_batteryPower->powerInterconnectionLoss, 0.0, error);
+
+    check_net_flows("5th case");
+
+
+    // charging happens from pv
+    m_batteryPower->powerBatteryDC = -20;
+    m_batteryPowerFlow->calculate();
+
+    EXPECT_NEAR(m_batteryPower->powerBatteryAC, -20.4, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToLoad, 1.62, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToBattery, 20.40, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToBattery, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToLoad, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToGrid, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerBatteryToLoad, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerConversionLoss, 2.39, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemLoss, 1.0, error);
+    EXPECT_NEAR(m_batteryPower->powerCritLoadUnmet, 48.39, error);
+    EXPECT_NEAR(m_batteryPower->powerInterconnectionLoss, 0.0, error);
+
+    check_net_flows("6th case");
+
+    // excess discharging causes interconnection loss
+    m_batteryPower->powerBatteryDC = 50;
+    m_batteryPowerFlow->calculate();
+
+    EXPECT_NEAR(m_batteryPower->powerBatteryAC, 26.33, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToLoad, 23.67, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToBattery, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToBattery, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerGridToLoad, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemToGrid, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerBatteryToLoad, 26.32, error);
+    EXPECT_NEAR(m_batteryPower->powerConversionLoss, 3.39, error);
+    EXPECT_NEAR(m_batteryPower->powerSystemLoss, 1.0, error);
+    EXPECT_NEAR(m_batteryPower->powerCritLoadUnmet, 0, error);
+    EXPECT_NEAR(m_batteryPower->powerInterconnectionLoss, 0.0, error);
+
+    check_net_flows("7th case");
```

Comment:
seems like the battery charging will not be reduced to reduce unmet load, and the battery will be curtailed to reduce interconnection loss. Is this always true?

Reply:
Yes to the first part - I view that as the job of dispatch code, not the powerflow code.

For the second part - that's only true during an outage. For an interconnection limit, PV's export to grid is curtailed first, followed by fuel cell, followed by battery discharging. However, P_battery_ac is not yet reduced for an interconnection limit, so there's work for dispatch to do there too.

Reply:
@brtietz, do we need to funnel in the grid interconnection limits to the dispatch code to account for additional adjustments to P_battery_ac?

Reply:
Yes, that is planned for a subsequent pull request.